pandas/core/dtypes/missing.py

```diff
@@ -321,6 +324,14 @@ def _isna_string_dtype(values: np.ndarray, inf_as_na: bool) -> npt.NDArray[np.bo
     return result
 
 
+def _isna_recarray_dtype(values: np.recarray, inf_as_na: bool) -> npt.NDArray[np.bool_]:
+    result = np.zeros(values.shape, dtype=bool)
+    for i, record in enumerate(values):
+        result[i] = libmissing.isnaobj(np.array(record), inf_as_na=inf_as_na)
```

Comment:
I don't think `isnaobj` is well defined for a record. Do all elements in the record need to be na-like? Does the record itself need to be na-like?

Reply:
@mroeschke What is your opinion on this?